controllers/azurecluster_controller.go

```diff
@@ -296,3 +297,44 @@ func (acr *AzureClusterReconciler) reconcileDelete(ctx context.Context, clusterS
 
 	return reconcile.Result{}, nil
 }
+
+func clusterToInfrastructureMapFunc(ctx context.Context, c client.Client) handler.MapFunc {
+	log := ctrl.LoggerFrom(ctx)
```

Comment:
in CAPZ we have a logger used for tracing, see #1806 

```suggestion
	ctx, log, done := tele.StartSpanWithLogger(ctx, "controllers.AzureClusterReconciler.clusterToInfrastructureMapFunc")
	defer done()
```